content/docs/intro/concepts/programming-model.md

```diff
@@ -2624,9 +2624,11 @@ Any outputs can be returned by your `create` function in the `outs` property of
 
 > **Note:** The following only applies to statically typed languages.
 
-If you need to access the outputs of your custom resource outside it with strong typing support, declare each output property returned in the `outs` property by your `create` function as a class member of the `pulumi.dynamic.Resource` itself. For example, in TypeScript, these must be declared as `public readonly` class members in your `pulumi.dynamic.Resource` class. These class members must also have the type `pulumi.Output<T>`.
+If you need to access the outputs of your custom resource outside it with strong typing support, declare each output property returned in the `outs` property by your `create` function as a class member of the `pulumi.dynamic.Resource` itself. The name of the class member must match the names of the output properties as returned by the `create` function.
 
-> **Note:** The name of the class member must match the names of the output properties as returned by the `create` function.
+For example, in TypeScript, these must be declared as `public readonly` class members in your `pulumi.dynamic.Resource` class. These class members must also have the type `pulumi.Output<T>`.
+
+> **Note:** If `--strictPropertyInitialization` is enabled (it typically is enabled by default along with other `--strict` mode flags), [definite assignment assertion modifiers](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#strict-class-initialization) (an `!` exclamation point) must be used on each class member to relay to TypeScript that these will be initialized indirectly.
```

Comment:
Do you think we need this?  I worry about having anything overly TypeScript specific in the prose - as it distracts from others.  If we have the code sample show the right thing - is that enough?

Reply:
OK, I'll cut it. We already have a lot of overly TypeScript specific prose, so erring on the side of not adding more, and reducing it in general, sounds good to me.